gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -955,3 +955,60 @@ hook.Add("TTTEndRound", "TTTEndRound4TTT2TargetPlayer", function()
 		plys[i].targetPlayer = nil
 	end
 end)
+
+---
+-- Returns if the player is reviving.
+-- @return boolean The blocking status
+-- @realm shared
+function plymeta:IsReviving()
+	return self.isReviving or false
+end
+
+---
+-- Returns if the ongoing revival is blocking or not.
+-- @return boolean The blocking status
+-- @realm shared
+function plymeta:IsBlockingRevival()
+	return self.isBlockingRevival or false
+end
+
+---
+-- Returns the time when the ongoing revival started.
+-- @return [default=@{CurTime()}]number The time when the revival started in seconds
+-- @realm shared
+function plymeta:GetRevivalStartTime()
+	return self.revivalStartTime or CurTime()
+end
+
+---
+-- Returns the duration for the ongoing revival.
+-- @return [default=1.0]number The time for the revival in seconds
+-- @realm shared
+function plymeta:GetRevivalDuration()
+	return self.revivalDurarion or 0.0
+end
+
+---
+-- Checks if a player was active during this round. A player was active if they received
+-- a role. This state is reset once the next round begins (@{GM:TTTBeginRound}).
+-- @return boolean Returns if the player was active
+-- @realm shared
+function plymeta:WasActiveInRound()
+	return self:TTT2NETGetBool("player_was_active_in_round", false)
+end
+
+---
+-- Checks if a player died while the round was active.
+-- @return boolean Returns if the player died in the round
+-- @realm shared
+function plymeta:HasDiedInRound()
+	return self:TTT2NETGetBool("player_has_died_in_round", false)
+end
+
+---
+-- Checks if a player was revived in the round.
+-- @return boolean Returns if the player was revived
+-- @realm shared
+function plymeta:WasRevivedInRound()
+	return self:HasDiedInRound() and self:IsTerror()
```

Comment:
The second check isn't needed and would lead to the wrong result. A died terror is in spectator team.

Reply:
It should only return true if a player was revived in the round, hece this check. Otherwise it would still return true if the player is dead. Which is what `HasDiedInRound()` does.

Reply:
But based in the function name, A dead terror (spectator) could also have been revived in round...

Reply:
I'll have to check my intentions for this again and I'll reconsider the name and contents afterwars.